tests/test_search.py

```diff
@@ -304,3 +304,27 @@ def test_parallel(app):
     app.build()
     index = load_searchindex(app.outdir / 'searchindex.js')
     assert index['docnames'] == ['index', 'nosearch', 'tocitem']
+
+
+@pytest.mark.sphinx(testroot='search')
+def test_search_index_is_deterministic(app):
+    app.builder.build_all()
+    index = load_searchindex(app.outdir / 'searchindex.js')
+
+    def check_if_dicts_are_sorted(item):
+        if type(item) is dict:
+            keys = list(item.keys())
+            if not all(keys[i] <= keys[i + 1] for i in range(len(keys) - 1)):
+                msg = f'keys are not sorted: {item!r}'
```

Comment:
I'd prefer having an assertion to check rather a RuntimeError (pytest should catch it I think?). And have the assert message directly (no msg variable)

Reply:
The msg variable was [a complaint ruff made](https://github.com/sphinx-doc/sphinx/actions/runs/6047168044/job/16410100144), but I'll change the code soon to use an assertion.